include/deal.II/grid/grid_tools.h

```diff
@@ -170,6 +170,19 @@ namespace GridTools
   template <int dim, typename T>
   double cell_measure (const T &, ...);
 
+  /**
+   * Project a point to Triangulation Iterator object.
+   *
+   * Returns the closest point to @p trial_point that lies on the given iterator object.
+   *
+   * @author Luca Heltai, 2017.
+   */
+  template <typename Iterator>
+  Point<Iterator::AccessorType::space_dimension>
+  project_to_object(const Iterator &object,
+                    const Point<Iterator::AccessorType::space_dimension> &trial_point);
```

Comment:
the description of this function would suggest that it finds the closest point on the object, which one could interpret as "the object as described by its vertices and the associated manifold(s)". But it doesn't work with the manifold at all -- it just considers the d-linear approximation. I think this should be reflected in the documentation, or even in the name.

Reply:
This function has been redone to use the Manifold information.

Reply:
I think the first sentence is still a bit confusing. It says "Project a point to Triangulation Iterator object.", but it's not immediately clear to me what exactly this object is. Presumably it something to do with the manifold associated with the object, but even then it's not completely clear to me. Is something like this correct?

"Project a point to an object described by the provided Triangulation Iterator. For example, if the iterator points to a one-dimensional line or edge, then the the returned point will be a point on the geometric line that connects the vertices as the manifold associated with the object sees it (i.e., the geometric line may be curved if it lives in a higher dimensional space). If the iterator points to a quadrilateral in a higher dimensional space, then the returned point lies within the convex hull of the vertices of the quad as seen by the associated manifold."